File: Python/svr.py
#SVM
#Carga de Librerias
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#Carga de Dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:,2].values

#Escalado de Variables
from sklearn.preprocessing import StandardScaler
sc_X = StandardScaler()
sc_y = StandardScaler()
X = sc_X.fit_transform(X)
y = sc_y.fit_transform(y.reshape(-1,1))

#SVR
from sklearn.svm import SVR
regression= SVR(kernel= "rbf")
regression.fit(X, y)
y_pred = sc_y.inverse_transform(regression.predict(sc_X.transform([[6.5]])))

plt.scatter (X, y,  color = "red")
plt.plot(X, regression.predict(X), color = "blue")
plt.title("Modelo de Regresion (SVR)")
plt.xlabel("Posicion de empleado")
plt.ylabel("Sueldo")
plt.show()

File: R/svr.R
# SVR
#Carga de Dataset
dataset = read.csv('Position_Salaries.csv')
dataset = dataset[,2:3]

#Carga de Libreria e1071 para usar SVM
#install.packages("e1071")
library(e1071)

#Modelo SVM
regression = svm(formula = Salary ~ .,
                 data = dataset,
                 type = "eps-regression",
                 kernel= "radial")

#Prediccion con Nivel de 6.5 ¿Cuanto debería de ganar?
y_pred = predict(regression, newdata= data.frame(Level =6.5))
library(ggplot2)

#Grafico de SVM
x_grid = seq(min(dataset$Level), max(dataset$Level),0.1)
ggplot()+
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             color = "red")+
  geom_line(aes(x = dataset$Level, y = predict(regression, newdata = data.frame(Level= dataset$Level))),
            color ="blue")+
  ggtitle("Prediccion SVM")+
  xlab("Nivel del empleado")+
  ylab("Salario")

